subsys/net/lib/lwm2m/lwm2m_obj_swmgmt.c

```diff
@@ -0,0 +1,743 @@
+/*
+ * Copyright (c) 2020 Voi Technologies AB
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define LOG_MODULE_NAME net_lwm2m_obj_swmgmt
+#define LOG_LEVEL CONFIG_LWM2MM_LOG_LEVEL_DBG
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#include <string.h>
+#include <init.h>
+
+#include <stdint.h>
+#include <net/lwm2m.h>
+
+#include "lwm2m_object.h"
+#include "lwm2m_engine.h"
+
+#include "lwm2m_pull_context.h"
+
+#define SWMGMT_PACKAGE_NAME_ID 0
+#define SWMGMT_PACKAGE_VERSION_ID 1
+#define SWMGMT_PACKAGE_ID 2
+#define SWMGMT_PACKAGE_URI_ID 3
+#define SWMGMT_INSTALL_ID 4
+#define SWMGMT_CHECKPOINT_ID 5
+#define SWMGMT_UNINSTALL_ID 6
+#define SWMGMT_UPDATE_STATE_ID 7
+#define SWMGMT_UPDATE_SUPPORTED_OBJECTS_ID 8
+#define SWMGMT_UPDATE_RESULT_ID 9
+#define SWMGMT_ACTIVATE_ID 10
+#define SWMGMT_DEACTIVATE_ID 11
+#define SWMGMT_ACTIVATION_UPD_STATE_ID 12
+#define SWMGMT_PACKAGE_SETTINGS_ID 13
+#define SWMGMT_USER_NAME_ID 14
+#define SWMGMT_PASSWORD_ID 15
+#define SWMGMT_MAX_ID 16
+
+#define PACKAGE_NAME_LEN 255
+#define PACKAGE_VERSION_LEN 255
+#define PACKAGE_URI_LEN 255
```

Comment:
It's a lot of space for the package name/version, needed per object instance. I think it'd be good to make those values configurable with Kconfig.

Reply:
done and I'm setting some lower default values